src/core/numerics/ion_updater/ion_updater.hpp

```diff
@@ -259,9 +259,13 @@ void IonUpdater<Ions, Electromag, GridLayout>::updateAndDepositAll_(Ions& ions,
                                                    interpolator_, layout, inGhostBox, inGhostLayer);
 
             auto& particleArray = particleRange.array();
-            particleArray.export_particles(
-                domainParticles, [&](auto const& cell) { return isIn(Point{cell}, domainBox); });
-
+            {
+                auto enteredInDomain
```

Comment:
I am not sure this is what you think it is.
Given the name `enteredInDomain` it appears you mean it is the range or ghost particles that have entered the domain after their move.
However, and if I understand correctly, this is the range start:pivot from the inDomainBox() partition that is performed on the ghost range `start:InGhostLayer.iend()` but `inGhostLayer` is the range of ghost particles that have stayed in the ghost **layer** (not box), i.e. not in domain, not outside ghost box.

see : 

```cpp 
            return isIn(Point{cell}, ghostBox) and !isIn(Point{cell}, domainBox);
```

therefore I don't think that any particle in that range can actually be found in domain, resulting in `enteredInDomain` range always being empty and no copy happens.

am I right?

Reply:
btw, if I am right, the fact that it passes the tests is concerning, in particular for such an annoying-to-debug part of the code, we may be smart to have a better test of that to prevent from merging crap leading to nasty bugs a year from now :)

Reply:
well, I think it's strange that we find any particles *in domain* from such a range, as it seems